src/rust/cryptography-x509-verification/src/policy/mod.rs

```diff
@@ -247,103 +243,96 @@ impl<'a, B: CryptoOps> Policy<'a, B> {
-                // 5280 4.2.1.8: Subject Directory Attributes
-                ExtensionPolicy::maybe_present(
-                    SUBJECT_DIRECTORY_ATTRIBUTES_OID,
-                    Criticality::NonCritical,
-                    None,
-                ),
+            ca_extension_policy: ExtensionPolicy {
                 // 5280 4.2.2.1: Authority Information Access
-                ExtensionPolicy::maybe_present(
-                    AUTHORITY_INFORMATION_ACCESS_OID,
+                authority_information_access: ExtensionValidator::maybe_present(
                     Criticality::NonCritical,
                     Some(common::authority_information_access),
                 ),
-            ]),
-            ca_extension_policies: Vec::from([
                 // 5280 4.2.1.1: Authority Key Identifier
-                ExtensionPolicy::maybe_present(
-                    AUTHORITY_KEY_IDENTIFIER_OID,
+                authority_key_identifier: ExtensionValidator::maybe_present(
                     Criticality::NonCritical,
                     Some(ca::authority_key_identifier),
                 ),
                 // 5280 4.2.1.2: Subject Key Identifier
                 // NOTE: CABF requires SKI in CA certificates, but many older CAs lack it.
                 // We choose to be permissive here.
-                ExtensionPolicy::maybe_present(
-                    SUBJECT_KEY_IDENTIFIER_OID,
+                subject_key_identifier: ExtensionValidator::maybe_present(
                     Criticality::NonCritical,
                     None,
                 ),
                 // 5280 4.2.1.3: Key Usage
-                ExtensionPolicy::present(KEY_USAGE_OID, Criticality::Agnostic, Some(ca::key_usage)),
+                key_usage: ExtensionValidator::present(Criticality::Agnostic, Some(ca::key_usage)),
+                subject_alternative_name: ExtensionValidator::maybe_present(
+                    Criticality::Agnostic,
+                    None,
+                ),
                 // 5280 4.2.1.9: Basic Constraints
-                ExtensionPolicy::present(
-                    BASIC_CONSTRAINTS_OID,
+                basic_constraints: ExtensionValidator::present(
                     Criticality::Critical,
                     Some(ca::basic_constraints),
                 ),
                 // 5280 4.2.1.10: Name Constraints
                 // NOTE: MUST be critical in 5280, but CABF relaxes to MAY.
-                ExtensionPolicy::maybe_present(
-                    NAME_CONSTRAINTS_OID,
+                name_constraints: ExtensionValidator::maybe_present(
                     Criticality::Agnostic,
                     Some(ca::name_constraints),
                 ),
                 // 5280 4.2.1.11: Policy Constraints
-                ExtensionPolicy::maybe_present(POLICY_CONSTRAINTS_OID, Criticality::Critical, None),
+                policy_constraints: ExtensionValidator::maybe_present(Criticality::Critical, None),
```

Comment:
Thinking about it a bit, this should also probably have a validator function that rejects this extension outright -- 5280 says that it's always critical (which we check) which means that we must handle it, which we intentionally don't.

Reply:
(I'll build a testcase that'll work for that.)